services/web-app/utils/mdx-image-optimizer-plugin.mjs

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { getPlaiceholder } from 'plaiceholder'
+import { visit } from 'unist-util-visit'
+
+const mdxImageOptimizerPlugin = (dirPath) => async (tree) => {
+  const matches = []
+  visit(tree, { type: 'image' }, (node) => matches.push(node))
+
+  const promises = matches.map(async (node) => {
+    // construct full image path (with repo url)
+    const fullPath = `${dirPath}/${node.url}?raw=true`
```

Comment:
Would this break if you encounter an image that already has a fully qualified URL as its source?

Also, does the allow you to do something like `../../../../my-different-repo/some-vulnerable-image-i-want-to-display-in-everyones-browser.png` by specifying a maliciously crafted URL in your markdown?

Better question: Can we enforce that the source markdown file and the target org/repo combo is the same when we've determined that the image is coming from github?

Reply:
> added check to not alter url if image has a fully qualified url
> added check to validate repo hasn't changed when constructing absolute url from relative source